src/components/search/search-dropdown.vue

```diff
@@ -0,0 +1,733 @@
+<script setup lang="ts">
+import { ref, watch, nextTick } from 'vue'
+import { useTranslation } from 'i18next-vue'
+import { fetchApi } from '@/services/api/api.service'
+import { storeToRefs } from 'pinia'
+import { searchLayerService } from '@/services/search/search-layer.service'
+import { useMapStore } from '@/stores/map.store'
+import { matchCoordinate } from '@/services/search/coordinate.service'
+import { useThemeStore } from '@/stores/config.store'
+import useLayers from '@/composables/layers/layers.composable'
+import useBackgroundLayer from '@/composables/background-layer/background-layer.composable'
+import useThemes from '@/composables/themes/themes.composable'
+import FilterPanel from './filter-panel.vue'
+import { curFilters, esMatch } from '@/composables/search/search-filters'
+import { transformExtent } from 'ol/proj.js'
+import useMap from '@/composables/map/map.composable'
+
+const { findThemeNamesByLayerId } = useThemes()
+const { t, i18next } = useTranslation()
+const { maxZoom } = storeToRefs(useMapStore())
+// Define reactive variables
+const searchQuery = ref('')
+const isOpenResults = ref(false)
+const searchResults = ref<
+  {
+    header: string
+    selectResult: Function
+    results: {
+      label: string
+      layer_name: string
+      entry: object
+      showRoutingButton: boolean
+    }[]
+  }[]
+>([])
+const isLoading = ref(false) // Track loading state
+const layerLookup: { [key: string]: string[] } = {
+  Adresse: ['addresses'],
+  Parcelle: ['parcels', 'parcels_labels'],
+  Parcelle_go: ['parcels_go', 'parcels_labels_go'],
+  Parcelle_prof: ['parcels_prof', 'parcels_labels'],
+  lieu_dit: ['toponymes'],
+  FLIK: ['asta_flik_parcels'],
+  FLIK_Provisoire: ['asta_flik_parcels_provisoire'],
+  asta_esp: ['asta_esp_esp'],
+  nom_de_rue: ['roads', 'roads_labels'],
+}
+
+function addLayerFromSearch(layer_name: string) {
+  const { theme } = useThemeStore()
+  let cur_suggestion_layer: string = layer_name
+
+  if (
+    layer_name === 'Parcelle' &&
+    typeof theme === 'string' &&
+    ['go', 'prof'].includes(theme)
+  ) {
+    cur_suggestion_layer = cur_suggestion_layer + '_' + theme
+  }
+
+  var layers = layerLookup[cur_suggestion_layer] || []
+  const { findByName } = useThemes()
+  layers.forEach(function (layer) {
+    const layerToAdd = findByName(layer)
+    if (layerToAdd !== undefined) {
+      useLayers().toggleLayer(layerToAdd.id, true, false, false)
```

Comment:
maybe extract `toggleLayer` somewhere up in the code? this would prevent doing `useLayers()` at each loop iteraction?